catalog/opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/impl/TestOpenSearchSource.java

```diff
@@ -681,6 +746,47 @@ private String getSample() {
         return "<xml></xml>";
     }
 
+    protected SecureCxfClientFactory getMockFactory(WebClient client) {
+        SecureCxfClientFactory factory = mock(SecureCxfClientFactory.class);
+
+        doReturn(client).when(factory)
+                .getClient();
+        doReturn(client).when(factory)
+                .getWebClientForSubject(any(org.apache.shiro.subject.Subject.class));
+        doReturn(client).when(factory)
+                .getWebClient();
+
+        return factory;
+    }
+
+    /**
+     * Example of a real-world foreign markup consumer.
+     */
+    private static class ForeignMarkupConsumerExample
+            implements BiConsumer<List<Element>, SourceResponse> {
+
+        private Long totalResults;
+
+        @Override
+        public void accept(List<Element> elements, SourceResponse sourceResponse) {
+            for (Element element : elements) {
+                if (element.getName()
+                        .equals("totalResults")) {
+                    try {
+                        totalResults = Long.parseLong(element.getContent(0)
```

Comment:
❓ It doesn't really matter since it's just a unit test/sample, but should this be summing the `totalResults` from each element instead of taking the last value?

Reply:
In the case of totalResults, there would only be one element of that name.